migration_original/ODS1Stage/tables/Show/SOLRPracticeDelta/spu_translated_SOLRPracticeDelta.sql

```diff
@@ -0,0 +1,198 @@
+                                from
+                                    Raw.ProviderDeltaProcessing as pdp
+                                    inner join Base.ProviderToOffice as pto on pdp.ProviderID = pto.ProviderID
+                                    inner join Base.Office as o on pto.OfficeID = o.OfficeID
+                                WHERE
+                                    o.practiceid not in (
+                                        select
+                                            practiceid
+                                        from
+                                            Show.SOLRPracticeDelta
+                                    )) as source
+                                ON source.Practiceid = target.Practiceid
+                                WHEN NOT MATCHED THEN 
+                                INSERT (
+                                PracticeID,
+                                SolrDeltaTypeCode,
+                                StartDeltaProcessDate,
+                                MidDeltaProcessComplete
+                                )
+                                VALUES (
+                                source.PracticeID,
+                                source.SolrDeltaTypeCode,
+                                source.StartDeltaProcessDate,
+                                source.MidDeltaProcessComplete
+                                );$$;
+
+        EXECUTE IMMEDIATE truncate_statement;
+        EXECUTE IMMEDIATE merge_statement_2;
+        EXECUTE IMMEDIATE merge_statement_3;
+           
+    END IF;
+
+
+---------------------------------------------------------
+----------------- 3. SQL Statements ---------------------
+---------------------------------------------------------     
+
+
+
+---------------------------------------------------------
+--------- 4. Actions (Inserts and Updates) --------------
+---------------------------------------------------------  
+
+
+merge_statement_4 := 'MERGE INTO Show.SOLRPracticeDelta as target USING 
+                   (SELECT DISTINCT
+                        PracticeId,
+                        StartDeltaProcessDate,
+                        EndDeltaProcessDate
+                   FROM Show.SOLRPracticeDelta
+                   WHERE StartDeltaProcessDate IS NOT NULL AND EndDeltaProcessDate IS NULL) as source 
+                   ON source.Practiceid = target.Practiceid
+                   WHEN MATCHED THEN
+                    UPDATE
+                    SET 
+                        target.EndDeltaProcessDate  = CURRENT_TIMESTAMP()';
+                   
+---------------------------------------------------------
+------------------- 5. Execution ------------------------
+--------------------------------------------------------- 
+
+-- EXECUTE IMMEDIATE truncate_statement;
+-- EXECUTE IMMEDIATE merge_statement_1;
+-- EXECUTE IMMEDIATE merge_statement_2;
+-- EXECUTE IMMEDIATE merge_statement_3;
+EXECUTE IMMEDIATE merge_statement_4 ;
```

Comment:
How about moving the IF ELSE to this section?

Reply:
Done!